nameresolution/configuration.go

```diff
@@ -0,0 +1,48 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package nameresolution
+
+import (
+	"fmt"
+)
+
+// ConvertConfig converts map[interface{}]interface{} to map[string]interface{} to normalize
+// for JSON and usage in component initialization.
+func ConvertConfig(i interface{}) (interface{}, error) {
```

Comment:
Can this be moved into dapr/kit? Even better, is there any other existing utility that does that?

Reply:
@artursouza I added this, and thought about putting it there. Do we know of anywhere else this is useful other than name resolution? In general, I like the idea of putting it in kit anyway. Can we proceed with merging this PR and I'll take an action item to move it to kit along with some other config stuff I am working on?

Reply:
There are definitely projects out there that deal with this.  [Here is a thread with some history](https://github.com/go-yaml/yaml/issues/139). In general, its probably not worth pulling in a new module just for this 1 function.

I'm thinking we should eventually move this to daprd so that the config comes pre converted/cleaned up. Again baby steps. I'll take this on as I move more config related stuff to kit.

Reply:
OK.